compilation failure on skybridge

on skybridge:

Currently Loaded Modules:
  1) gnu/10.2.1   2) cmake/3.20.3   3) openmpi-gnu/4.0

I get the compilation error below with:

cmake \
-D CMAKE_C_COMPILER:PATH="mpicc" \
-D CMAKE_CXX_COMPILER:PATH="mpicxx" \
-D PARTHENON_ENABLE_TESTING=TRUE \
-D PARTHENON_IMPORT_KOKKOS=OFF \
-D PARTHENON_ENABLE_REGRESSION_TESTS=OFF \
-D PARTHENON_DISABLE_EXAMPLES=OFF \
..

/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/boundary_conditions.cpp:22:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc_in_one.cpp:31:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/flux_correction_cc.cpp:36:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/fc/flux_correction_fc.cpp:34:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/fc/bvals_fc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:76: src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:160: src/CMakeFiles/parthenon.dir/bvals/bvals.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:118: src/CMakeFiles/parthenon.dir/bvals/fc/bvals_fc.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:132: src/CMakeFiles/parthenon.dir/bvals/fc/flux_correction_fc.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:104: src/CMakeFiles/parthenon.dir/bvals/cc/flux_correction_cc.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:146: src/CMakeFiles/parthenon.dir/bvals/boundary_conditions.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:90: src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc_in_one.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1646: src/CMakeFiles/parthenon.dir/all] Error 2

I updated it again to today (2-22-22) with develop and I see the same issue.

The build script:

```
#!/bin/bash

# need when cloned:
# git submodule update --init

export HDF5_DIR="$HOME/packages/hdf5/build/hdf5"
export HDF5_ROOT="$HOME/packages/hdf5/build/hdf5"

cmake \
-D CMAKE_C_COMPILER:PATH="mpicc" \
-D CMAKE_CXX_COMPILER:PATH="mpicxx" \
-D PARTHENON_ENABLE_TESTING=TRUE \
-D PARTHENON_IMPORT_KOKKOS=OFF \
-D PARTHENON_ENABLE_REGRESSION_TESTS=OFF \
-D PARTHENON_DISABLE_EXAMPLES=OFF \
..

cmake --build . -j 8
```

Currently Loaded Modules:
  1) gnu/10.2.1   2) openmpi-gnu/4.0   3) cmake/3.20.3

All the output:

```
../../build_parth.sh 
CMake Warning at cmake/MachineCfg.cmake:61 (message):
  Not using any machine configuration.  Consider creating a configuration
  file following the examples in /cmake/machine_cfgs/ and then point the
  MACHINE_CFG cmake or environment variable to your custom file.Note, that
  the machine file can be placed in any directory (also outside the repo).
Call Stack (most recent call first):
  CMakeLists.txt:17 (include)


-- The C compiler identification is GNU 10.2.1
-- The CXX compiler identification is GNU 10.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/openmpi/4.0/gnu/bin/mpicc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/openmpi/4.0/gnu/bin/mpicxx - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at cmake/Format.cmake:49 (message):
  Couldn't determine clang-format version.  clang-format 8.0 is required -
  results on other versions may not be stable
Call Stack (most recent call first):
  CMakeLists.txt:65 (include)


-- Setting build type to 'RelWithDebInfo' as none was specified.
-- Found MPI_CXX: /opt/openmpi/4.0/gnu/bin/mpicxx (found version "3.1") 
-- Found MPI: TRUE (found version "3.1") found components: CXX 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5") found components: CXX 
-- Found HDF5: /ascldap/users/brtnfld/packages/hdf5/build/hdf5/lib/libhdf5.so;/ascldap/users/brtnfld/packages/szip-2.1/szip/lib/libsz.so;/usr/lib64/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so (found version "1.13.1-1") found components: C 
-- Setting default Kokkos CXX standard to 14
-- Setting policy CMP0074 to use <Package>_ROOT variables
-- The project name is: Kokkos
-- Using -std=c++14 for C++14 standard as feature
-- Built-in Execution Spaces:
--     Device Parallel: NoTypeDefined
--     Host Parallel: Kokkos::OpenMP
--       Host Serial: NONE
-- 
-- Architectures:
-- Found TPLLIBDL: /usr/lib64/libdl.so  
-- Kokkos Devices: OPENMP, Kokkos Backends: OPENMP
-- Using Kokkos source from Parthenon submodule at /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos
-- Building unit tests.
-- Found Python3: /usr/bin/python3.6 (found version "3.6.8") found components: Interpreter 
-- Building integration tests.
-- Building performance tests.
-- PAR_LOOP_LAYOUT='SIMDFOR_LOOP' (default par_for wrapper layout)
-- PAR_LOOP_INNER_LAYOUT='SIMDFOR_INNER_LOOP' (default par_for_inner wrapper layout)
-- Found Git: /usr/bin/git (found version "2.30.1") 
-- Configuring done
-- Generating done
-- Build files have been written to: /ascldap/users/brtnfld/packages/parthenon.master/build
Consolidate compiler generated dependencies of target kokkoscore
[ 13%] Built target kokkoscore
Consolidate compiler generated dependencies of target kokkoscontainers
[ 16%] Built target kokkoscontainers
Consolidate compiler generated dependencies of target catch2_define
[ 18%] Built target catch2_define
Consolidate compiler generated dependencies of target parthenon
[ 20%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc_in_one.cpp.o
[ 20%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o
[ 20%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/cc/flux_correction_cc.cpp.o
[ 23%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/fc/bvals_fc.cpp.o
[ 23%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/fc/flux_correction_fc.cpp.o
[ 23%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/boundary_conditions.cpp.o
[ 25%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/bvals.cpp.o
[ 25%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/bvals_base.cpp.o
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/boundary_conditions.cpp:22:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc_in_one.cpp:31:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/bvals.cpp:39:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/flux_correction_cc.cpp:36:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/bvals_base.cpp:32:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/fc/flux_correction_fc.cpp:34:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/fc/bvals_fc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:146: src/CMakeFiles/parthenon.dir/bvals/boundary_conditions.cpp.o] Error 1
gmake[2]: *** Waiting for unfinished jobs....
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:104: src/CMakeFiles/parthenon.dir/bvals/cc/flux_correction_cc.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:118: src/CMakeFiles/parthenon.dir/bvals/fc/bvals_fc.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:132: src/CMakeFiles/parthenon.dir/bvals/fc/flux_correction_fc.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:160: src/CMakeFiles/parthenon.dir/bvals/bvals.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:76: src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:174: src/CMakeFiles/parthenon.dir/bvals/bvals_base.cpp.o] Error 1
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:90: src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc_in_one.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:1646: src/CMakeFiles/parthenon.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2



```

I tried without the wrappers and did a verbose build:

```
CMake Warning at cmake/MachineCfg.cmake:61 (message):
  Not using any machine configuration.  Consider creating a configuration
  file following the examples in /cmake/machine_cfgs/ and then point the
  MACHINE_CFG cmake or environment variable to your custom file.Note, that
  the machine file can be placed in any directory (also outside the repo).
Call Stack (most recent call first):
  CMakeLists.txt:17 (include)


-- The C compiler identification is GNU 10.2.1
-- The CXX compiler identification is GNU 10.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/rh/devtoolset-10/root/usr/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/rh/devtoolset-10/root/usr/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at cmake/Format.cmake:49 (message):
  Couldn't determine clang-format version.  clang-format 8.0 is required -
  results on other versions may not be stable
Call Stack (most recent call first):
  CMakeLists.txt:65 (include)


-- Setting build type to 'RelWithDebInfo' as none was specified.
-- Found MPI_CXX: /opt/openmpi/4.0/gnu/lib/libmpi.so (found version "3.1") 
-- Found MPI: TRUE (found version "3.1") found components: CXX 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5") found components: CXX 
-- Found HDF5: /ascldap/users/brtnfld/packages/hdf5/build/hdf5/lib/libhdf5.so;/ascldap/users/brtnfld/packages/szip-2.1/szip/lib/libsz.so;/usr/lib64/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so (found version "1.13.1-1") found components: C 
-- Setting default Kokkos CXX standard to 14
-- Setting policy CMP0074 to use <Package>_ROOT variables
-- The project name is: Kokkos
-- Using -std=c++14 for C++14 standard as feature
-- Built-in Execution Spaces:
--     Device Parallel: NoTypeDefined
--     Host Parallel: Kokkos::OpenMP
--       Host Serial: NONE
-- 
-- Architectures:
-- Found TPLLIBDL: /usr/lib64/libdl.so  
-- Kokkos Devices: OPENMP, Kokkos Backends: OPENMP
-- Using Kokkos source from Parthenon submodule at /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos
-- Building unit tests.
-- Found Python3: /usr/bin/python3.6 (found version "3.6.8") found components: Interpreter 
-- Building integration tests.
-- Building performance tests.
-- PAR_LOOP_LAYOUT='SIMDFOR_LOOP' (default par_for wrapper layout)
-- PAR_LOOP_INNER_LAYOUT='SIMDFOR_INNER_LOOP' (default par_for_inner wrapper layout)
-- Found Git: /usr/bin/git (found version "2.30.1") 
-- Configuring done
-- Generating done
-- Build files have been written to: /ascldap/users/brtnfld/packages/parthenon.master/build
/projects/cmake/toss3/3.20.3/bin/cmake -P /ascldap/users/brtnfld/packages/parthenon.master/build/CMakeFiles/VerifyGlobs.cmake
/projects/cmake/toss3/3.20.3/bin/cmake -S/ascldap/users/brtnfld/packages/parthenon.master -B/ascldap/users/brtnfld/packages/parthenon.master/build --check-build-system CMakeFiles/Makefile.cmake 0
/projects/cmake/toss3/3.20.3/bin/cmake -E cmake_progress_start /ascldap/users/brtnfld/packages/parthenon.master/build/CMakeFiles /ascldap/users/brtnfld/packages/parthenon.master/build//CMakeFiles/progress.marks
/opt/rh/devtoolset-10/root/usr/bin/gmake  -f CMakeFiles/Makefile2 all
gmake[1]: Entering directory '/home/brtnfld/packages/parthenon.master/build'
/opt/rh/devtoolset-10/root/usr/bin/gmake  -f Kokkos/core/src/CMakeFiles/kokkoscore.dir/build.make Kokkos/core/src/CMakeFiles/kokkoscore.dir/depend
gmake[2]: Entering directory '/home/brtnfld/packages/parthenon.master/build'
cd /ascldap/users/brtnfld/packages/parthenon.master/build && /projects/cmake/toss3/3.20.3/bin/cmake -E cmake_depends "Unix Makefiles" /ascldap/users/brtnfld/packages/parthenon.master /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src /ascldap/users/brtnfld/packages/parthenon.master/build /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src/CMakeFiles/kokkoscore.dir/DependInfo.cmake --color=
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
/opt/rh/devtoolset-10/root/usr/bin/gmake  -f Kokkos/core/src/CMakeFiles/kokkoscore.dir/build.make Kokkos/core/src/CMakeFiles/kokkoscore.dir/build
gmake[2]: Entering directory '/home/brtnfld/packages/parthenon.master/build'
[  0%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_CPUDiscovery.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_CPUDiscovery.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_CPUDiscovery.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_CPUDiscovery.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_CPUDiscovery.cpp
[  2%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Core.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Core.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_Core.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_Core.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_Core.cpp
[  2%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Error.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Error.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_Error.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_Error.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_Error.cpp
[  2%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_ExecPolicy.cpp
[  4%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostBarrier.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostBarrier.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_HostBarrier.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_HostBarrier.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_HostBarrier.cpp
[  4%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_HostSpace.cpp
[  4%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace_deepcopy.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace_deepcopy.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace_deepcopy.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace_deepcopy.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_HostSpace_deepcopy.cpp
[  6%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostThreadTeam.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_HostThreadTeam.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_HostThreadTeam.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_HostThreadTeam.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_HostThreadTeam.cpp
[  6%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_MemoryPool.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_MemoryPool.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_MemoryPool.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_MemoryPool.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_MemoryPool.cpp
[  6%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_MemorySpace.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_MemorySpace.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_MemorySpace.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_MemorySpace.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_MemorySpace.cpp
[  9%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_NumericTraits.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_NumericTraits.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_NumericTraits.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_NumericTraits.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_NumericTraits.cpp
[  9%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Profiling.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Profiling.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_Profiling.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_Profiling.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_Profiling.cpp
[  9%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Serial_Task.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Serial_Task.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_Serial_Task.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_Serial_Task.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_Serial_Task.cpp
[ 11%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_SharedAlloc.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_SharedAlloc.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_SharedAlloc.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_SharedAlloc.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_SharedAlloc.cpp
[ 11%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Spinwait.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Spinwait.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_Spinwait.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_Spinwait.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_Spinwait.cpp
[ 11%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Stacktrace.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_Stacktrace.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_Stacktrace.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_Stacktrace.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_Stacktrace.cpp
[ 11%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_hwloc.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/impl/Kokkos_hwloc.cpp.o -MF CMakeFiles/kokkoscore.dir/impl/Kokkos_hwloc.cpp.o.d -o CMakeFiles/kokkoscore.dir/impl/Kokkos_hwloc.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/impl/Kokkos_hwloc.cpp
[ 13%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Exec.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Exec.cpp.o -MF CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Exec.cpp.o.d -o CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Exec.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/OpenMP/Kokkos_OpenMP_Exec.cpp
[ 13%] Building CXX object Kokkos/core/src/CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Task.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/core/src/CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Task.cpp.o -MF CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Task.cpp.o.d -o CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Task.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src/OpenMP/Kokkos_OpenMP_Task.cpp
[ 13%] Linking CXX static library libkokkoscore.a
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /projects/cmake/toss3/3.20.3/bin/cmake -P CMakeFiles/kokkoscore.dir/cmake_clean_target.cmake
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src && /projects/cmake/toss3/3.20.3/bin/cmake -E cmake_link_script CMakeFiles/kokkoscore.dir/link.txt --verbose=1
/opt/rh/devtoolset-10/root/usr/bin/ar qc libkokkoscore.a CMakeFiles/kokkoscore.dir/impl/Kokkos_CPUDiscovery.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_Core.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_Error.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_ExecPolicy.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_HostBarrier.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_HostSpace_deepcopy.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_HostThreadTeam.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_MemoryPool.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_MemorySpace.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_NumericTraits.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_Profiling.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_Serial_Task.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_SharedAlloc.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_Spinwait.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_Stacktrace.cpp.o CMakeFiles/kokkoscore.dir/impl/Kokkos_hwloc.cpp.o CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Exec.cpp.o CMakeFiles/kokkoscore.dir/OpenMP/Kokkos_OpenMP_Task.cpp.o
/opt/rh/devtoolset-10/root/usr/bin/ranlib libkokkoscore.a
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
[ 13%] Built target kokkoscore
/opt/rh/devtoolset-10/root/usr/bin/gmake  -f Kokkos/containers/src/CMakeFiles/kokkoscontainers.dir/build.make Kokkos/containers/src/CMakeFiles/kokkoscontainers.dir/depend
gmake[2]: Entering directory '/home/brtnfld/packages/parthenon.master/build'
cd /ascldap/users/brtnfld/packages/parthenon.master/build && /projects/cmake/toss3/3.20.3/bin/cmake -E cmake_depends "Unix Makefiles" /ascldap/users/brtnfld/packages/parthenon.master /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/containers/src /ascldap/users/brtnfld/packages/parthenon.master/build /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src/CMakeFiles/kokkoscontainers.dir/DependInfo.cmake --color=
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
/opt/rh/devtoolset-10/root/usr/bin/gmake  -f Kokkos/containers/src/CMakeFiles/kokkoscontainers.dir/build.make Kokkos/containers/src/CMakeFiles/kokkoscontainers.dir/build
gmake[2]: Entering directory '/home/brtnfld/packages/parthenon.master/build'
[ 16%] Building CXX object Kokkos/containers/src/CMakeFiles/kokkoscontainers.dir/impl/Kokkos_UnorderedMap_impl.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT Kokkos/containers/src/CMakeFiles/kokkoscontainers.dir/impl/Kokkos_UnorderedMap_impl.cpp.o -MF CMakeFiles/kokkoscontainers.dir/impl/Kokkos_UnorderedMap_impl.cpp.o.d -o CMakeFiles/kokkoscontainers.dir/impl/Kokkos_UnorderedMap_impl.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/containers/src/impl/Kokkos_UnorderedMap_impl.cpp
[ 16%] Linking CXX static library libkokkoscontainers.a
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src && /projects/cmake/toss3/3.20.3/bin/cmake -P CMakeFiles/kokkoscontainers.dir/cmake_clean_target.cmake
cd /ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src && /projects/cmake/toss3/3.20.3/bin/cmake -E cmake_link_script CMakeFiles/kokkoscontainers.dir/link.txt --verbose=1
/opt/rh/devtoolset-10/root/usr/bin/ar qc libkokkoscontainers.a CMakeFiles/kokkoscontainers.dir/impl/Kokkos_UnorderedMap_impl.cpp.o
/opt/rh/devtoolset-10/root/usr/bin/ranlib libkokkoscontainers.a
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
[ 16%] Built target kokkoscontainers
/opt/rh/devtoolset-10/root/usr/bin/gmake  -f tst/CMakeFiles/catch2_define.dir/build.make tst/CMakeFiles/catch2_define.dir/depend
gmake[2]: Entering directory '/home/brtnfld/packages/parthenon.master/build'
cd /ascldap/users/brtnfld/packages/parthenon.master/build && /projects/cmake/toss3/3.20.3/bin/cmake -E cmake_depends "Unix Makefiles" /ascldap/users/brtnfld/packages/parthenon.master /ascldap/users/brtnfld/packages/parthenon.master/tst /ascldap/users/brtnfld/packages/parthenon.master/build /ascldap/users/brtnfld/packages/parthenon.master/build/tst /ascldap/users/brtnfld/packages/parthenon.master/build/tst/CMakeFiles/catch2_define.dir/DependInfo.cmake --color=
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
/opt/rh/devtoolset-10/root/usr/bin/gmake  -f tst/CMakeFiles/catch2_define.dir/build.make tst/CMakeFiles/catch2_define.dir/build
gmake[2]: Entering directory '/home/brtnfld/packages/parthenon.master/build'
[ 18%] Building CXX object tst/CMakeFiles/catch2_define.dir/catch2_define.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/tst && /opt/rh/devtoolset-10/root/usr/bin/g++ -DCATCH_CONFIG_ENABLE_BENCHMARKING -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/external/Catch2/single_include -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/algorithms/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/algorithms/src -O2 -g -DNDEBUG -fopenmp -std=c++14 -MD -MT tst/CMakeFiles/catch2_define.dir/catch2_define.cpp.o -MF CMakeFiles/catch2_define.dir/catch2_define.cpp.o.d -o CMakeFiles/catch2_define.dir/catch2_define.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/tst/catch2_define.cpp
[ 18%] Linking CXX static library libcatch2_define.a
cd /ascldap/users/brtnfld/packages/parthenon.master/build/tst && /projects/cmake/toss3/3.20.3/bin/cmake -P CMakeFiles/catch2_define.dir/cmake_clean_target.cmake
cd /ascldap/users/brtnfld/packages/parthenon.master/build/tst && /projects/cmake/toss3/3.20.3/bin/cmake -E cmake_link_script CMakeFiles/catch2_define.dir/link.txt --verbose=1
/opt/rh/devtoolset-10/root/usr/bin/ar qc libcatch2_define.a CMakeFiles/catch2_define.dir/catch2_define.cpp.o
/opt/rh/devtoolset-10/root/usr/bin/ranlib libcatch2_define.a
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
[ 18%] Built target catch2_define
/opt/rh/devtoolset-10/root/usr/bin/gmake  -f src/CMakeFiles/parthenon.dir/build.make src/CMakeFiles/parthenon.dir/depend
gmake[2]: Entering directory '/home/brtnfld/packages/parthenon.master/build'
cd /ascldap/users/brtnfld/packages/parthenon.master/build && /projects/cmake/toss3/3.20.3/bin/cmake -E cmake_depends "Unix Makefiles" /ascldap/users/brtnfld/packages/parthenon.master /ascldap/users/brtnfld/packages/parthenon.master/src /ascldap/users/brtnfld/packages/parthenon.master/build /ascldap/users/brtnfld/packages/parthenon.master/build/src /ascldap/users/brtnfld/packages/parthenon.master/build/src/CMakeFiles/parthenon.dir/DependInfo.cmake --color=
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
/opt/rh/devtoolset-10/root/usr/bin/gmake  -f src/CMakeFiles/parthenon.dir/build.make src/CMakeFiles/parthenon.dir/build
gmake[2]: Entering directory '/home/brtnfld/packages/parthenon.master/build'
[ 20%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/src/generated -I/ascldap/users/brtnfld/packages/hdf5/build/hdf5/include -I/ascldap/users/brtnfld/packages/szip-2.1/szip/include -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/algorithms/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/algorithms/src -isystem /opt/openmpi/4.0/gnu/include -O2 -g -DNDEBUG -pthread -fopenmp -std=c++14 -MD -MT src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o -MF CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o.d -o CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:103:16: error: redefinition of ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’
  103 | inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
      |                ^~~~~~~~~~~~~~~~~~~~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:97:15: note: ‘T parthenon::Env::Impl::parse_value(std::string&) [with T = long unsigned int; std::string = std::basic_string<char>]’ previously declared here
   97 | inline size_t parse_value<size_t>(std::string &strvalue) {
      |               ^~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:79: src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o] Error 1
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
gmake[1]: *** [CMakeFiles/Makefile2:1649: src/CMakeFiles/parthenon.dir/all] Error 2
gmake[1]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
gmake: *** [Makefile:149: all] Error 2
```

With the new branch:

```
[ 14%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/src/generated -I/ascldap/users/brtnfld/packages/hdf5/build/hdf5/include -I/ascldap/users/brtnfld/packages/szip-2.1/szip/include -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/algorithms/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/algorithms/src -isystem /opt/openmpi/4.0/gnu/include -O2 -g -DNDEBUG -pthread -fopenmp -std=c++14 -MD -MT src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o -MF CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o.d -o CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:109:39: error: ‘T’ was not declared in this scope
  109 | template <std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
      |                                       ^
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:109:49: error: template argument 1 is invalid
  109 | template <std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
      |                                                 ^
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:109:63: error: template argument 1 is invalid
  109 | template <std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
      |                                                               ^
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:110:49: error: ‘T’ was not declared in this scope
  110 |           typename std::enable_if<!std::is_same<T, size_t>::value, bool>::type = true>
      |                                                 ^
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:110:58: error: template argument 1 is invalid
  110 |           typename std::enable_if<!std::is_same<T, size_t>::value, bool>::type = true>
      |                                                          ^
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:110:72: error: template argument 1 is invalid
  110 |           typename std::enable_if<!std::is_same<T, size_t>::value, bool>::type = true>
      |                                                                        ^
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:110:73: error: ‘<declaration error>’ is not a template [-fpermissive]
  110 |           typename std::enable_if<!std::is_same<T, size_t>::value, bool>::type = true>
      |                                                                         ^~
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:111:8: error: ‘T’ does not name a type
  111 | inline T parse_value(std::string &strvalue) {
      |        ^
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:79: src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o] Error 1
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
gmake[1]: *** [CMakeFiles/Makefile2:1649: src/CMakeFiles/parthenon.dir/all] Error 2
gmake[1]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
gmake: *** [Makefile:149: all] Error 2

```

```
 14%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/src/generated -I/ascldap/users/brtnfld/packages/hdf5/build/hdf5/include -I/ascldap/users/brtnfld/packages/szip-2.1/szip/include -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/algorithms/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/algorithms/src -isystem /opt/openmpi/4.0/gnu/include -O2 -g -DNDEBUG -pthread -fopenmp -std=c++14 -MD -MT src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o -MF CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o.d -o CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:114:2: error: invalid preprocessing directive #Endif; did you mean #endif?
  114 | #Endif // ifdef ENABLE_HDF5
      |  ^~~~~
      |  endif
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:17: error: unterminated #ifndef
   17 | #ifndef UTILS_UTILS_HPP_
      | 
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/mesh_data.hpp:31,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:40,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:17: error: unterminated #ifndef
   17 | #ifndef UTILS_UTILS_HPP_
      | 
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:109:78: error: ‘std::enable_if<std::is_same<T, long unsigned int>::value, bool>::type’ is not a type
  109 | template <typename T, std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
      |                                                                              ^~~~
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:79: src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o] Error 1
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
gmake[1]: *** [CMakeFiles/Makefile2:1649: src/CMakeFiles/parthenon.dir/all] Error 2
gmake[1]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
gmake: *** [Makefile:149: all] Error 2
```

```
[ 14%] Building CXX object src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o
cd /ascldap/users/brtnfld/packages/parthenon.master/build/src && /opt/rh/devtoolset-10/root/usr/bin/g++ -DKOKKOS_DEPENDENCE -I/ascldap/users/brtnfld/packages/parthenon.master/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/src/generated -I/ascldap/users/brtnfld/packages/hdf5/build/hdf5/include -I/ascldap/users/brtnfld/packages/szip-2.1/szip/include -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/core/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/containers/src -I/ascldap/users/brtnfld/packages/parthenon.master/build/Kokkos/algorithms/src -I/ascldap/users/brtnfld/packages/parthenon.master/external/Kokkos/algorithms/src -isystem /opt/openmpi/4.0/gnu/include -O2 -g -DNDEBUG -pthread -fopenmp -std=c++14 -MD -MT src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o -MF CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o.d -o CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o -c /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp
In file included from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_device_context.hpp:17,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm_boundaries.hpp:16,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/swarm.hpp:37,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/interface/state_descriptor.hpp:29,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/application_input.hpp:24,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/mesh/mesh.hpp:33,
                 from /ascldap/users/brtnfld/packages/parthenon.master/src/bvals/cc/bvals_cc.cpp:37:
/ascldap/users/brtnfld/packages/parthenon.master/src/utils/utils.hpp:109:78: error: ‘std::enable_if<std::is_same<T, long unsigned int>::value, bool>::type’ is not a type
  109 | template <typename T, std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
      |                                                                              ^~~~
gmake[2]: *** [src/CMakeFiles/parthenon.dir/build.make:79: src/CMakeFiles/parthenon.dir/bvals/cc/bvals_cc.cpp.o] Error 1
gmake[2]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
gmake[1]: *** [CMakeFiles/Makefile2:1649: src/CMakeFiles/parthenon.dir/all] Error 2
gmake[1]: Leaving directory '/home/brtnfld/packages/parthenon.master/build'
gmake: *** [Makefile:149: all] Error 2

```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -15,6 +15,7 @@
 - [[PR 617]](https://github.com/lanl/parthenon/pull/617) Unify the coordinates API for MeshBlockPack and VariablePack
 
 ### Fixed (not changing behavior/API/variables/...)
+- [[PR 652]](https://github.com/lanl/parthenon/pull/652) Fix issue with hsize_t and size_t in utils parser
 - [[PR 649]](https://github.com/lanl/parthenon/pull/649) Ensure LoadBalancing send buffers are filled and allow async recv LB
 - [[PR 618]](https://github.com/lanl/parthenon/pull/618) Fix bug in variable pack performance test
 - [[PR 616]](https://github.com/lanl/parthenon/pull/609) Restore sparse base names in PackIndexMap

--- a/src/utils/utils.hpp
+++ b/src/utils/utils.hpp
@@ -24,6 +24,7 @@
 #include <cstdint>
 #include <sstream>
 #include <string>
+#include <type_traits>
 
 #include "constants.hpp"
 #include "error_checking.hpp"
@@ -62,7 +63,12 @@ namespace Env {
 
 namespace Impl {
 // TODO(the person bumping standard to C++17) Clean up this mess and use constexpr if
-template <typename T>
+template <typename T,
+          typename std::enable_if<!std::is_same<T, size_t>::value, bool>::type = true
+#ifdef ENABLE_HDF5
+          ,
+          typename std::enable_if<!std::is_same<T, hsize_t>::value, bool>::type = true>
+#endif
 T parse_value(std::string &strvalue);
 
 // Parse env. variable expected to hold a bool value allowing for different conventions.
@@ -93,14 +99,16 @@ T parse_unsigned(const std::string &strvalue) {
   return res;
 }
 
-template <>
-inline size_t parse_value<size_t>(std::string &strvalue) {
+template <typename T,
+          typename std::enable_if<std::is_same<T, size_t>::value, bool>::type = true>
+inline T parse_value(std::string &strvalue) {
   return parse_unsigned<size_t>(strvalue);
 }
 
 #ifdef ENABLE_HDF5
-template <>
-inline hsize_t parse_value<hsize_t>(std::string &strvalue) {
+template <std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
+          typename std::enable_if<!std::is_same<T, size_t>::value, bool>::type = true>
+inline T parse_value(std::string &strvalue) {
   return parse_unsigned<hsize_t>(strvalue);
 }
 #endif // ifdef ENABLE_HDF5

--- a/src/utils/utils.hpp
+++ b/src/utils/utils.hpp
@@ -63,6 +63,21 @@ namespace Env {
 
 namespace Impl {
 // TODO(the person bumping standard to C++17) Clean up this mess and use constexpr if
+
+// TODO(JMM): We use template specialization below to parse
+// information from environment variables
+// where we template on return type. We hit an issue with HDF5's hsize_t, becuase:
+// (a) sometimes HDF5 is disabled, so hsize_t isn't always present
+// (b) hsize_t's type is system/compiler-dependent. Sometimes it's the same as size_t,
+//     but it doesn't have to be. If you define a specialization for size_t
+//     and hsize_t on a system where they are not the same,
+//     the compiler will see a duplicate function and complain.
+// To resolve this issue, we use SFINAE. The std::enable_ifs mean the
+// templated functions are enabled only if the appropriate conditions are met.
+// In particular, the prototype is enabled only for types that are
+// not size_t or hsize_t.
+// Then the implementation for size_t is always enabled and the implementation
+// for hsize_t is enabled ONLY if HDF5 is available, and hsize_t != size_t.
 template <typename T,
           typename std::enable_if<!std::is_same<T, size_t>::value, bool>::type = true
 #ifdef ENABLE_HDF5

--- a/src/utils/utils.hpp
+++ b/src/utils/utils.hpp
@@ -106,7 +106,8 @@ inline T parse_value(std::string &strvalue) {
 }
 
 #ifdef ENABLE_HDF5
-template <typename T, std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
+template <typename T,
+          typename std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
           typename std::enable_if<!std::is_same<T, size_t>::value, bool>::type = true>
 inline T parse_value(std::string &strvalue) {
   return parse_unsigned<hsize_t>(strvalue);

--- a/src/utils/utils.hpp
+++ b/src/utils/utils.hpp
@@ -106,12 +106,12 @@ inline T parse_value(std::string &strvalue) {
 }
 
 #ifdef ENABLE_HDF5
-template <std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
+template <typename T, std::enable_if<std::is_same<T, hsize_t>::value, bool>::type = true,
           typename std::enable_if<!std::is_same<T, size_t>::value, bool>::type = true>
 inline T parse_value(std::string &strvalue) {
   return parse_unsigned<hsize_t>(strvalue);
 }
-#endif // ifdef ENABLE_HDF5
+#Endif // ifdef ENABLE_HDF5
 } // namespace Impl
 
 // Get environment variables of various types (with checks).

--- a/src/utils/utils.hpp
+++ b/src/utils/utils.hpp
@@ -68,7 +68,7 @@ template <typename T,
 #ifdef ENABLE_HDF5
           ,
           typename std::enable_if<!std::is_same<T, hsize_t>::value, bool>::type = true>
-#endif
+#endif // ENABLE_HDF5
 T parse_value(std::string &strvalue);
 
 // Parse env. variable expected to hold a bool value allowing for different conventions.
@@ -111,7 +111,7 @@ template <typename T, std::enable_if<std::is_same<T, hsize_t>::value, bool>::typ
 inline T parse_value(std::string &strvalue) {
   return parse_unsigned<hsize_t>(strvalue);
 }
-#Endif // ifdef ENABLE_HDF5
+#endif // ifdef ENABLE_HDF5
 } // namespace Impl
 
 // Get environment variables of various types (with checks).

--- a/src/utils/utils.hpp
+++ b/src/utils/utils.hpp
@@ -3,7 +3,7 @@
 // Copyright(C) 2014 James M. Stone <jmstone@princeton.edu> and other code contributors
 // Licensed under the 3-clause BSD License, see LICENSE file for details
 //========================================================================================
-// (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+// (C) (or copyright) 2020-2022. Triad National Security, LLC. All rights reserved.
 //
 // This program was produced under U.S. Government contract 89233218CNA000001 for Los
 // Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC

